nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -779,6 +774,11 @@ in {
           '';
           serviceConfig.Type = "oneshot";
           serviceConfig.User = "nextcloud";
+          serviceConfig.LoadCredential = lib.flatten [
+            "adminpass:${cfg.config.adminpassFile}"
+            (lib.optional (cfg.config.dbpassFile != null) "dbpass:${cfg.config.dbpassFile}")
+            (lib.optional (cfg.config.objectstore.s3.enable) "s3:${cfg.config.objectstore.s3.secretFile}")
+          ];
```

Comment:
Rather than duplicating the value of `LoadCredential` for several services, how about declaring it once at the top?

Also, how about

``` nix
[ "adminpass:${cfg.config.adminpassFile}" ]
++ lib.optional (cfg.config.dbpassFile != null) "dbpass:${cfg.config.dbpassFile}"
```